go/libkb/upak_lite.go

```diff
@@ -0,0 +1,273 @@
+package libkb
+
+import (
+	"fmt"
+	"io/ioutil"
+	"time"
+
+	"github.com/buger/jsonparser"
+	keybase1 "github.com/keybase/client/go/protocol/keybase1"
+)
+
+func LoadUPAKLite(arg LoadUserArg) (ret *keybase1.UpkLitev1AllIncarnations, err error) {
+	uid := arg.uid
+	m := arg.m
+
+	// get sig hints in order to populate during merkle leaf lookup
+	sigHints, err := LoadSigHints(m, arg.uid)
+	if err != nil {
+		return nil, err
+	}
+	leaf, err := lookupMerkleLeaf(m, uid, false, sigHints)
+	if err != nil {
+		return nil, err
+	}
+	// This might be a little heavy-handed, especially for big users.
+	// Perhaps we can get what we need by requesting less data from
+	// the server than going through user/lookup.
+	user, err := LoadUserFromServer(m, uid, nil)
+	if err != nil {
+		return nil, err
+	}
+	loader := NewHighSigChainLoader(m, user, leaf)
+	highChain, err := loader.Load()
+	if err != nil {
+		return nil, err
+	}
+	return highChain.ToUPAKLite(user)
+}
+
+type HighSigChainLoader struct {
+	BaseSigChainLoader
+	MetaContextified
+	user      *User
+	leaf      *MerkleUserLeaf
+	chain     *HighSigChain
+	chainType *ChainType
+	links     ChainLinks
+	ckf       ComputedKeyFamily
+	dirtyTail *MerkleTriple
+}
+
+type HighSigChain struct {
+	BaseSigChain
+	Contextified
+	uid                  keybase1.UID
+	username             NormalizedUsername
+	chainLinks           ChainLinks
+	localCki             *ComputedKeyInfos
```

Comment:
will we need these `local` attrs if we're only going to use highsigchain for other users?

Reply:
i believe you are correct. will remove this.